libpod/runtime_img.go

```diff
@@ -268,12 +271,27 @@ func (r *Runtime) LoadImage(ctx context.Context, name, inputFile string, writer
 			newImages, err = r.ImageRuntime().LoadFromArchiveReference(ctx, src, signaturePolicy, writer)
 		}
 		if err != nil {
-			src, err := directory.NewReference(inputFile)
-			if err == nil {
-				newImages, err = r.ImageRuntime().LoadFromArchiveReference(ctx, src, signaturePolicy, writer)
+			if strings.Contains(name, "/") {
+				// prepend "localhost/" to support local image saved with this semantics
+				src, err = ociarchive.NewReference(inputFile, filepath.Join("localhost", name))
+				if err == nil {
+					newImages, err = r.ImageRuntime().LoadFromArchiveReference(ctx, src, signaturePolicy, writer)
+				}
 			}
 			if err != nil {
-				return "", errors.Wrapf(err, "error pulling %q", name)
+				src, err := directory.NewReference(inputFile)
+				if err == nil {
+					newImages, err = r.ImageRuntime().LoadFromArchiveReference(ctx, src, signaturePolicy, writer)
+				}
+				if err != nil {
+					src, err := layout.NewReference(inputFile, name)
+					if err == nil {
+						newImages, err = r.ImageRuntime().LoadFromArchiveReference(ctx, src, signaturePolicy, writer)
+					}
+					if err != nil {
+						return "", errors.Wrapf(err, "error pulling %q", name)
+					}
+				}
```

Comment:
BTW this series of ever-deeper-nesting `if`s is very repetitive and fairly hard to read, especially if there were any copy&paste errors it would be difficult to notice.

Please consider consolidating this, either by building a `[]types.ImageReference` in advance (for all possible cases, skipping ones that fail) and iterating over it, or maybe
```go
for _, referenceFn := []range func() (types.ImageReference, error) {
    func() (
        types.ImageReference, error) { return dockerarchive.ParseReference(inputFile) // FIXME? We should add dockerarchive.NewReference()
    },
    …
} {
   ref, err := referenceFn()
   if err == nil {
      if newImages, err = …; err == nil {
          return …
       }
   }
}